astropy/coordinates/sky_coordinate.py

```diff
@@ -1485,7 +1485,31 @@ def radial_velocity_correction(self, kind='barycentric', obstime=None,
            rv = rv + vcorr + rv * vcorr / consts.c
 
         If your target is nearby and/or has finite proper motion you may need to account
-        for terms arising from this. See Wright & Eastmann (2014) for details.
+        for terms arising from this. See Wright & Eastman (2014) for details.
+
+        Also note that this method returns the correction velocity in the so-called
+        *optical convention*::
+
+           vcorr = zb * consts.c
+
+        where `zb` is the barycentric correction redshift as defined in section 3
+        of Wright & Eastman (2014). The application formula given above follows from their
+        equation (11) under assumption that the radial velocity `rv` has also been defined
+        using the same optical convention. Note, this can be regarded as a matter of
+        velocity definition and does not by itself imply any loss of accuracy, provided
+        sufficient care has been taken during interpretation of the results. If you need
+        the barycentric correction expressed as the full relativistic velocity (e.g., to provide
+        it as the input to another software which performs the application), the
+        following recipe can be used::
+
+           zb = vcorr / consts.c
+           zb_plus_one_squared = (zb + 1) * (zb + 1)
```

Comment:
Is there benefit using this over `(zb + 1) ** 2`?

Reply:
in the c-world such an explicit multiplication would be faster. In python, honestly I don't know. Anyway, I am happy to change it if it adds to readability

Reply:
In the Python world, it is very easy to find out. And you appear to be correct! 😄 
```python
In [1]: zb = 0.05

In [2]: %timeit  (zb + 1) * (zb + 1)
83.2 ns ± 0.291 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)

In [3]: %timeit  (zb + 1) ** 2
144 ns ± 0.577 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)
```

Reply:
Go for readability. Generally `zb` will be an array and numpy is smart enough to turn `**2` into `np.square` (likely even in-place).

Plus, the calculations before this are *far* slower.